akka-http/src/main/scala/akka/http/scaladsl/coding/Deflate.scala

```diff
@@ -89,10 +89,19 @@ private[http] object DeflateCompressor {
 class DeflateDecompressor(maxBytesPerChunk: Int = Decoder.MaxBytesPerChunkDefault) extends DeflateDecompressorBase(maxBytesPerChunk) {
 
   override def createLogic(attr: Attributes) = new DecompressorParsingLogic {
-    override val inflater: Inflater = new Inflater()
+    var maybeNoWrappedInflater: Option[Inflater] = None
+    override def inflater: Inflater = maybeNoWrappedInflater.getOrElse(new Inflater())
 
     override val inflateState = new Inflate(true) {
       override def onTruncation(): Unit = completeStage()
+
+      override def parse(reader: ByteStringParser.ByteReader): ParseResult[ByteString] = {
+        val data = reader.remainingData
+        maybeNoWrappedInflater = maybeNoWrappedInflater
+          .orElse(data.headOption.filter(b ⇒ (b & 0x0F) != 8)
```

Comment:
We should probably add a `peekByte` method to `ByteReader`. This then needs to be documented with all the information we researched here in this thread.

Reply:
Possibly - ByteReader is from Akka Streams though - does that not need a seperate issue/PR?